awsauth.go

```diff
@@ -172,15 +175,52 @@ func (this *Credentials) expired() bool {
 	}
 }
 
-type metadata struct {
-	algorithm       string
-	credentialScope string
-	signedHeaders   string
-	date            string
-	region          string
-	service         string
+type Metadata struct {
+	Algorithm       string
+	CredentialScope string
+	SignedHeaders   string
+	Date            string
+	Region          string
+	Service         string
+}
+
+func (m *Metadata) SetAlgorithm(value string){
```

Comment:
I see that these setter functions prevent any original value from being overwritten--what's the need for that constraint?